indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java

```diff
@@ -705,31 +650,28 @@ private void scheduleTasksCleanupForWorker(final String workerHostAndPort)
 
     final ListenableScheduledFuture<?> cleanupTask = cleanupExec.schedule(
         () -> {
-          log.info("Running scheduled cleanup for Worker[%s]", workerHostAndPort);
+          log.info("Running scheduled cleanup for worker[%s]", workerHostAndPort);
           try {
-            Set<HttpRemoteTaskRunnerWorkItem> tasksToFail = new HashSet<>();
-            synchronized (statusLock) {
-              for (Map.Entry<String, HttpRemoteTaskRunnerWorkItem> e : tasks.entrySet()) {
-                if (e.getValue().getState() == HttpRemoteTaskRunnerWorkItem.State.RUNNING) {
-                  Worker w = e.getValue().getWorker();
-                  if (w != null && w.getHost().equals(workerHostAndPort)) {
-                    tasksToFail.add(e.getValue());
-                  }
+            final Set<HttpRemoteTaskRunnerWorkItem> tasksToFail = new HashSet<>();
+            tasks.forEach((taskId, taskEntry) -> {
```

Comment:
Note that `.forEach()` since it is not mutually exclusive with `.compute()` and essentially just provides iteration over a snapshot. But I suppose it is okay here?